ci/linkcheck.sh

```diff
@@ -0,0 +1,21 @@
+#!/usr/bin/env bash
+
+if [ "$TRAVIS_EVENT_TYPE" = "cron" ] ; then # running in cron job
+  FLAGS=""
+
+  echo "Doing full link check."
+elif [ "$CI" = "true" ] ; then # running in PR CI build
+  COMMIT_RANGE=$(echo $TRAVIS_COMMIT_RANGE | sed 's/\.\.\./../g')
+  CHANGED_FILES=$(git diff-tree --no-commit-id --name-only -r $COMMIT_RANGE | tr '\n' ' ')
+  FLAGS="-f $CHANGED_FILES"
+
+  echo "Checking files changed in $TRAVIS_COMMIT_RANGE: $CHANGED_FILES"
+else # running locally
+  COMMIT_RANGE=master..
+  CHANGED_FILES=$(git diff-tree --no-commit-id --name-only -r $COMMIT_RANGE | tr '\n' ' ')
```

Comment:
```suggestion
  COMMIT_RANGE=origin/master...
  CHANGED_FILES=$(git diff --name-only $COMMIT_RANGE | tr '\n' ' ')
```

Reply:
`git diff-tree` doesn't seem to be working locally, while `git diff` is.

Reply:
Interesting... it looks like `git diff` parses ranges with different syntax from `git diff-tree`...

@camelid Any reason for including `origin/master`, rather than just `master`? For me, my local master branch is more up-to-date than my origin/master branch, so using `origin/master` makes it think that nearly every file has changed...

Reply:
Ah, I forgot that that's for the local build only. I'll update my suggestion :)